R/assocTestAggregate.R

```diff
@@ -1,252 +1,264 @@
-                      # user supplied weights
-                      weight <- getSnpVariable(gdsobj, weight.user)
-                      weight <- weight[keep]
-                      
-                      weight0 <- is.na(weight) | weight == 0
-                      if (any(weight0)) {
-                          keep <- !weight0
-                          var.info <- var.info[keep,,drop=FALSE]
-                          geno <- geno[,keep,drop=FALSE]
-                          n.obs <- n.obs[keep]
-                          freq <- freq[keep,,drop=FALSE]
-                          weight <- weight[keep]
-                      }
-                  }
-                  
-                  # number of variant sites
-                  n.site <- length(unique(var.info$variant.id))
-
-                  # number of alternate alleles
-                  n.alt <- sum(geno, na.rm=TRUE)
-                  
-                  # number of samples with observed alternate alleles > 0
-                  n.sample.alt <- sum(rowSums(geno, na.rm=TRUE) > 0)
-               
-                  res[[i]] <- data.frame(n.site, n.alt, n.sample.alt)
-                  res.var[[i]] <- cbind(var.info, n.obs, freq, weight)
-                  
-                  if (n.site > 0) {
-                      # mean impute missing values
-                      if (any(n.obs < nrow(geno))) {
-                          geno <- .meanImpute(geno, freq$freq)
-                      }
-
-                      # do the test
-                      assoc <- testVariantSet(null.model, G=geno, weights=weight, 
-                                              test=test, burden.test=burden.test, 
-                                              neig = neig, ntrace = ntrace,
-                                              rho=rho)
-                                              # pval.method=pval.method)
-                      res[[i]] <- cbind(res[[i]], assoc, stringsAsFactors=FALSE)
-                  }
-
-                  if (verbose & n.iter > 1 & i %% set.messages == 0) {
-                      message(paste("Iteration", i , "of", n.iter, "completed"))
-                  }
-                  i <- i + 1
-                  iterate <- GWASTools::iterateFilter(gdsobj)
-              }
-
-              res <- list(results=bind_rows(res), variantInfo=res.var)
-              .annotateAssoc(gdsobj, res)
-          })
+setGeneric("assocTestAggregate", function(gdsobj, ...) standardGeneric("assocTestAggregate"))
+
+.match.arg <- function(test) {
+    if (length(test) > 1) test <- NULL
+    match.arg(test, choices=c("Burden", "SKAT", "fastSKAT", "SMMAT", "fastSMMAT", "SKATO"))
+}
+
+setMethod("assocTestAggregate",
+          "SeqVarIterator",
+          function(gdsobj, null.model, AF.max=1,
+                   weight.beta=c(1,1), weight.user=NULL,
+                   test=c("Burden", "SKAT", "fastSKAT", "SMMAT", "SKATO"),
+                   burden.test=c("Score", "Wald", "BinomiRare","CMP"),

```

Comment:
I had planned to remove the `burden.test` argument. The plan is to always do the score test, and write a function that approximates the effect size, its standard error, and percent variance explained. I'm undecided at this point if those 3 values will always be included in the main output, or if it will be a post-processing function that users can apply to a subset of results. 

I'm thinking that "BinomiRare" and "CMP" could just be options to the `test` argument. Do you see any reason that that would not work?